components/esp_websocket_client/esp_websocket_client.c

```diff
@@ -1092,17 +1128,33 @@ int esp_websocket_client_send_text(esp_websocket_client_handle_t client, const c
     return esp_websocket_client_send_with_opcode(client, WS_TRANSPORT_OPCODES_TEXT, (const uint8_t *)data, len, timeout);
 }
 
+int esp_websocket_client_send_text_partial(esp_websocket_client_handle_t client, const char *data, int len, TickType_t timeout)
+{
+    return esp_websocket_client_send_with_exact_opcode(client, WS_TRANSPORT_OPCODES_TEXT, (const uint8_t *)data, len, timeout);
```

Comment:
Now, this is another potential problem, we're sending something with the "exact" opcode, but the last frame should contain `FIN` (which is apparently missing)

and I think it's actually missing even after https://github.com/espressif/esp-protocols/pull/461 merged

Reply:
sorry, the problem is actually here https://github.com/espressif/esp-protocols/pull/368/files#diff-53ead0f3896f79090b1ed1e79e1cca84ac4e505faa1bb76727ab856dabebc297L1092

Reply:
In order to mark the last frame I added `esp_websocket_client_send_fin` API. 
And documented for `esp_websocket_client_send_text_partial` 
`- To mark the end of fragmented data, you should use the 'esp_websocket_client_send_fin(...)' API. This sends a FIN frame.`
https://github.com/espressif/esp-protocols/blob/f62db5cfa2e64a27828a90c42ce92c2d2cb4ae89/components/esp_websocket_client/include/esp_websocket_client.h#L284C1-L289C128

Reply:
Plus we're not locking the client and perform the usually API parameters check?